app/src/main/java/com/keylesspalace/tusky/util/SettingsUtils.kt

```diff
@@ -0,0 +1,12 @@
+package com.keylesspalace.tusky.util
+
+const val MIN_PROXY_PORT = 0
+const val MAX_PROXY_PORT = 65535
+private val PROXY_RANGE = IntRange(MIN_PROXY_PORT, MAX_PROXY_PORT)
+
+fun isValidProxyPort(value: Any): Boolean = when (value) {
```

Comment:
Can you put this into ProxyPreferencesFragment please? I'd only create util files when something is used in multiple palces, otherwise it just makes code harder to understand if stuff is all over the place.

Reply:
I think it will be used in two places: 

1. user input validation (ProxyPreferencesFragment)
2. when retrieving values from settings to create the proxy itself. Theoretically the settings should never be invalid, but since we're deserializing data, I personally don't trust it. Also if this PR is accepted, there could be pre-existing invalid settings that we might be loading from a user's computer.

Reply:
It is in my local already :) I've got the custom dialog + error message stuff, OK button disabled if the port isn't valid so you can't even save an invalid port, but I need to write tests for it before updating the PR.